ios/fastlane/Fastfile

```diff
@@ -63,24 +64,64 @@ platform :ios do
       keychain_password: "",
       log_output: true
     )
+  end
+
+  lane :build_and_sign do |options|
     # pod install
     cocoapods(
       repo_update: true
     )
-    # Increase build number to avoid redundant errors (this is currently not committed)
-    increment_build_number
     # Build and Archive app
     gym(
       scheme: options[:scheme],
       workspace: workspace,
-      export_method: options[:type],
+      export_method: options[:method],
       clean: true,
       output_name: options[:scheme]+".ipa",
       export_team_id: team_id
     )
   end
 
-  lane :publish do |options|
+  # Generate AdHoc build and push to S3
+  lane :publish_adhoc do |options|
+    version = get_info_plist_value(
+      path: app_name+"/Info.plist",
```

Comment:
As I have it on my project, each build target has it's own info.plist. The only difference that I see is that they have a different `Bundle display name`. I think that the actual name that is displayed for the app is `Product Name` on the Build Settings.
Can you make sure that the `Bundle display name` in the Info.plist has `$(PRODUCT_NAME)` as value and that the correct name is displayed for the envs? testing it with one env that is not production.